conf/cassandra.yaml

```diff
@@ -992,6 +992,8 @@ compaction_throughput: 64MiB/s
 # are completely written, and used in place of the prior sstables for
 # any range that has been written. This helps to smoothly transfer reads 
 # between the sstables, reducing page cache churn and keeping hot rows hot
+# Set sstable_preemptive_open_interval to null for disabled which is equivalent to the sstable_preemptive_open_interval_in_mb
+# being negative
```

Comment:
Comment is a lot longer than the rest of the block, would you mind wrapping it early. I'm guessing we still refer to deprecated parameters in other comments and we can remove them if the deprecated settings are actually removed.

Reply:
How do you set to `null`? I tried 

`sstable_preemptive_open_interval: null`

but got this on startup.
```
Exception (org.apache.cassandra.exceptions.ConfigurationException) encountered during startup: Invalid yaml. Those properties [sstable_preemptive_open_interval] are not valid
ERROR [main] 2022-07-27 11:34:46,521 CassandraDaemon.java:896 - Exception encountered during startup: Invalid yaml. Those properties [sstable_preemptive_open_interval] are not valid
Invalid yaml. Those properties [sstable_preemptive_open_interval] are not valid
```

then
`sstable_preemptive_open_interval:`

and got
```Exception (org.apache.cassandra.exceptions.ConfigurationException) encountered during startup: Invalid yaml. Those properties [sstable_preemptive_open_interval] are not valid
ERROR [main] 2022-07-27 11:37:15,213 CassandraDaemon.java:896 - Exception encountered during startup: Invalid yaml. Those properties [sstable_preemptive_open_interval] are not valid
Invalid yaml. Those properties [sstable_preemptive_open_interval] are not valid```

and of course `-1` still fails as expected.

Reply:
Good catch, null is valid only if it is default value in the Config class which is not the case here :( 
I should have seen that. I will have to think about how to workaround it and add a startup test as now I double checked and it seems I made the same mistake yesterday with one more property so now I have to fix both of them. Thank you for catching it!

About the -1, it starts for me with sstable_preemptive_open_interval_in_mb = -1, then we see in the logs the converter loaded it as sstable_preemptive_open_interval = null; But I definitely null doesn't work.

Reply:
Thinking out loud.... we can do probably a mix of what you suggested initially and this fix for this special case. We can add a flag parameter to enable/disable which will be used with the new property if we want to disable it. If someone uses the old one we should prohibit setting the new flag then in parallel on startup if the old one was found in the yaml, WDYT?

Reply:
I might be able to workaround that need of additional flag actually and special case those two properties so that they do not error out when set to null.

Reply:
I think a separate flag would be a good idea and cleaner than relying on magic values to control whether features are enabled or not.  Does the new config framework have the capability to drive to settings from the deprecated `sstable_preemptive_open_interval_in_mb`?

Reply:
I already fixed it, but if we want a flag we can do that of course. The thing is the usage of null was accepted and documented long time ago, if we do it here differently, does this mean we need to change it in all other places we did use null?
In that case this would fix the issue with null and the VT as it seems that check is important for a work done in CASSANDRA-17166 as if I remove it VT throws currently null pointer exception. I have to check what @dcapwell did in that patch as I was not really working on that ticket. (CASSANDRA-17166)  

Looking back at the `Converters`, there are 4 other properties where we depend on `null`:
`credentials_update_interval, permissions_update_interval, roles_update_interval` - those are actually interesting in the context of this ticket as JMX uses the getters and returns their `*_validity` equivalent if they are null but VT shows null/-1... 
I will produce a patch for them separately. 
The 4-th one dependent on the null value (which I just fixed to be able to start with null, seems I didn't fully fix it the other day...) is `index_summary_resize_interval`. I will fix it after we agree which direction we prefer - flags or the null value. Now the only uncomfortable situation I see is that people will have to maintain two properties for one old one and if we continue adding feature flags and not removing properties, our config becomes too big and scary. 

`Does the new config framework have the capability to drive to settings from the deprecated sstable_preemptive_open_interval_in_mb?`

If you set the old one with `sstable_preemptive_open_interval_in_mb=-1` in cassandra.yaml, it loads internally through the `@Replaces` annotation and the `Converters` class into `sstable_preemptive_open_interval=null`. That is how the backward compatibility works on the parsing level for cassandra.yaml. I hope I didn't misunderstood your question.

Reply:
Oh, we actually have guardrails with null, as they are enabled per guardrail, the big guardrails feature flag was removed and they are individually enabled. Adding an enable flag for every guardrail will be a pain so I think we should stay with the null option, at least for those.

Reply:
The more I think, the more I am convinced enable/disable flags should work for features and not individual properties

Reply:
> I hope I didn't misunderstood your question.
I was asking if `sstable_preemptive_open_interval_in_mb` could set a new `sstable_preemptive_open_enabled` entry as well as `sstable_preemptive_open_interval`.

If you have a fix for setting `null` and don't need to create an `enabled` flag I'm fine with that. I thought you were saying you might need to create on to resolve this problem.

Reply:
`I was asking if sstable_preemptive_open_interval_in_mb could set a new sstable_preemptive_open_enabled entry as well as sstable_preemptive_open_interval.`
No, currently we map one or more old names to only one new property. 
`If you have a fix for setting null and don't need to create an enabled flag I'm fine with that. I thought you were saying you might need to create on to resolve this problem.`
I created a quick one by special casing this particular property to be able to set it to null even if the default value is not null. So I will need to figure out then only the VT Converters

Reply:
...and more tests to prevent us from future regressions of course :-)

Reply:
Oh, seems it was me not that ticket that broke it... null cannot be provided to `apply()`,  I will figure out something